From: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Subject: lustre: Simplify ib_post_(send|recv|srq_recv)() calls
Patch-mainline: not yet, under development
References: bsc#1103992 FATE#326009

Instead of declaring and passing a dummy 'bad_wr' pointer, pass NULL
as third argument to ib_post_(send|recv|srq_recv)().

Signed-off-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/staging/lustre/lnet/klnds/o2iblnd/o2iblnd_cb.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- a/drivers/staging/lustre/lnet/klnds/o2iblnd/o2iblnd_cb.c
+++ b/drivers/staging/lustre/lnet/klnds/o2iblnd/o2iblnd_cb.c
@@ -156,7 +156,6 @@ kiblnd_post_rx(struct kib_rx *rx, int cr
 {
 	struct kib_conn *conn = rx->rx_conn;
 	struct kib_net *net = conn->ibc_peer->ibp_ni->ni_data;
-	struct ib_recv_wr *bad_wrq = NULL;
 	int rc;
 
 	LASSERT(net);
@@ -188,10 +187,10 @@ kiblnd_post_rx(struct kib_rx *rx, int cr
 	 * own this rx (and rx::rx_conn) anymore, LU-5678.
 	 */
 	kiblnd_conn_addref(conn);
-	rc = ib_post_recv(conn->ibc_cmid->qp, &rx->rx_wrq, &bad_wrq);
+	rc = ib_post_recv(conn->ibc_cmid->qp, &rx->rx_wrq, NULL);
 	if (unlikely(rc)) {
-		CERROR("Can't post rx for %s: %d, bad_wrq: %p\n",
-		       libcfs_nid2str(conn->ibc_peer->ibp_nid), rc, bad_wrq);
+		CERROR("Can't post rx for %s: %d\n",
+		       libcfs_nid2str(conn->ibc_peer->ibp_nid), rc);
 		rx->rx_nob = 0;
 	}
 
@@ -847,8 +846,7 @@ kiblnd_post_tx_locked(struct kib_conn *c
 			 "bad wr_id %llx, opc %d, flags %d, peer: %s\n",
 			 bad->wr_id, bad->opcode, bad->send_flags,
 			 libcfs_nid2str(conn->ibc_peer->ibp_nid));
-		bad = NULL;
-		rc = ib_post_send(conn->ibc_cmid->qp, wrq, &bad);
+		rc = ib_post_send(conn->ibc_cmid->qp, wrq, NULL);
 	}
 
 	conn->ibc_last_send = jiffies;
